src/lib.rs

```diff
@@ -0,0 +1,325 @@
+        if unique_networks.len() != unreserved_networks.len() {
+            return Err("networks must be unique.".to_owned());
+        }
+
+        for account_type in &self.registry {
+            if let Some(network) = &account_type.network {
+                if network.chars().any(|c| c.is_whitespace()) {
+                    return Err(format!(
+                        "network can not have whitespace in: {:?}",
+                        account_type
+                    ));
+                }
+            }
+
+            if let Some(symbols) = &account_type.symbols {
+                if account_type
+                    .decimals
+                    .as_ref()
+                    .filter(|decimals| symbols.len() == decimals.len())
+                    .is_none()
+                {
+                    return Err(format!(
+                        "decimals must be specified for each symbol: {:?}",
+                        account_type
+                    ));
+                }
+            } else if account_type.decimals.is_some() {
+                return Err(format!(
+                    "decimals can't be specified without symbols: {:?}",
+                    account_type
+                ));
+            }
+
+            if let Some(sig_type) = &account_type.standard_account {
+                match sig_type.as_str() {
+                    "Sr25519"| "Ed25519" | "secp256k1" | "*25519" => {},
+                    _ => {
+                        return Err(format!("Unknown sig type in standardAccount: expected one of Sr25519, Ed25519, secp256k1, *25519: {:?}", account_type))
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+#[derive(Deserialize, Debug)]
+struct AccountType {
+    prefix: u16,
+    network: Option<String>,
+    #[serde(rename = "displayName")]
+    display_name: String,
+    #[serde(rename = "standardAccount")]
+    standard_account: Option<String>,
+    website: Option<String>,
+    symbols: Option<Vec<String>>,
+    decimals: Option<Vec<u8>>,
+}
+
+impl AccountType {
+    fn name(&self) -> String {
+        let mut name = self.network.clone().unwrap_or(
+            if let Some(standard_account) = &self.standard_account {
+                format!("Bare{}", standard_account)
+            } else {
+                let name = self
+                    .network
+                    .as_ref()
+                    .expect("network should not be empty if no account specified")
+                    .to_string();
+
+                assert!(
+                    name.starts_with("reserved"),
+                    "If no account specified, network should start `reserved` not:{}",
+                    name
+                );
+                name
+            },
+        );
+        let is_reserved = name.starts_with("reserved");
+        if name.ends_with("net") {
+            name.truncate(name.len() - 3);
+        }
```

Comment:
For what is this?

Reply:
The json and the enum names don't quite match. If we're happy changing the enum names then I will happily ditch this.

Reply:
The point is that the enum is generated out of the json and not make the enum look the same as the current enum.

Reply:
So please remove it